src/screencast/wlr_screencast.c

```diff
@@ -197,6 +202,8 @@ static void wlr_frame_failed(void *data,
 	logprint(TRACE, "wlroots: failed event handler");
 	cast->err = true;
 
+	xdpw_pwr_export_buffer(cast);
```

Comment:
At best, this will copy a bad frame in `writeFrameData`. Instead we should not perform the `memcpy`. I think it would be nice to set the `d[0].chunk->flags = SPA_CHUNK_FLAG_CORRUPTED` to indicate the buffer we enqueue to PipeWire has no meaningful contents (e.g. the v4l2 plugin does this, see `spa/plugins/v4l2/v4l2-utils.c`).

Reply:
flag was added, thanks for noticing! Needs to be added to clients to be respected.